interpreter/mesh/src/main/scala/io/buoyant/interpreter/MeshInterpreterInitializer.scala

```diff
@@ -0,0 +1,109 @@
+package io.buoyant.interpreter
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.conversions.time._
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.{H2, TlsClientPrep}
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.logging.Logger
+import io.buoyant.namer.{InterpreterConfig, InterpreterInitializer}
+import io.buoyant.interpreter.mesh.Client
+import scala.util.control.NoStackTrace
+
+/**
+ * The namerd interpreter offloads the responsibilities of name
+ * resolution to the namerd service via the namerd streaming gRPC API.
+ * Any namers configured in this linkerd are not used.
+ */
+class MeshInterpreterInitializer extends InterpreterInitializer {
+  val configClass = classOf[MeshInterpreterConfig]
+  override def configId: String = "io.l5d.mesh"
+}
+
+object MeshInterpreterInitializer extends MeshInterpreterInitializer
+
+object MeshInterpreterConfig {
+  private val log = Logger.get(getClass.getName)
+
+  val DefaultRoot = Path.Utf8("default")
+
+  val defaultRetry = Retry(1, 10.minutes.inSeconds)
+}
+
+case class Retry(
+  baseSeconds: Int,
+  maxSeconds: Int
+)
+
+case class MeshInterpreterConfig(
+  dst: Option[Path],
+  root: Option[Path],
+  tls: Option[MeshClientTlsConfig],
+  retry: Option[Retry]
+) extends InterpreterConfig {
+  import MeshInterpreterConfig._
+
+  @JsonIgnore
+  override val experimentalRequired = true
+
+  /**
+   * Construct a namer.
+   */
+  @JsonIgnore
+  def newInterpreter(params: Stack.Params): NameInterpreter = {
+    val name = dst match {
+      case None => throw new IllegalArgumentException("`dst` is a required field") with NoStackTrace
+      case Some(dst) => Name.Path(dst)
+    }
+    val label = MeshInterpreterInitializer.configId
+
+    val Retry(baseRetry, maxRetry) = retry.getOrElse(defaultRetry)
+    val backoffs = Backoff.exponentialJittered(baseRetry.seconds, maxRetry.seconds)
+
+    val client = H2.client
```

Comment:
I think we want retries on the client itself in addition to retries within the stream processing.  That way if the initial request fails, we can retry it.